File: ruby algos/atm.rb
def atm(cash_amount, denominations)
  bills_withdrawn = {}
  denominations.each do |denom|
    if denom <= cash_amount
      division, remainder = cash_amount.divmod(denom)
      bills_withdrawn[denom] = division
      cash_amount = remainder
    end
  end
  bills_withdrawn
end

p atm(110, [100, 20, 10, 5])
p atm(245, [20, 10, 5])

File: py algos/atm.py
def atm(amnt, denoms):
  wd = {}
  for denom in denoms:
    if denom <= amnt:
      val = amnt // denom
      wd[denom] = val
      amnt = amnt % denom
  return wd

print(atm(110, [20, 10, 5]))
print(atm(135, [20, 10, 5]))